### ISSUE TITLE
`emptyUpdate` got new semantics with opaleye-0.4

### ISSUE THREAD

**Issue by bergmark** [silkapp/girella#2]
To fix this I think we should scrap `runUpdateConst` and replace `emptyUpdate` with another TH-generated function `IdUpdate :: To Column R -> To Maybe (To Column R)` which could be used like this:

```
runUpdate table (\r -> (f r) { name = Just $ constant x }) condition
```

This will make sure that all code that uses it has to get updated.

**Comment by hesselink** [comment]
What changed exactly about the semantics of `emptyUpdate`?

**Comment by bergmark** [comment]
tomjaguarpaw/haskell-opaleye#92

**Comment by hesselink** [comment]
What would `runUpdate` do? Would it set all columns explicitly to their old values? I think that might be tricky, since they might have been updated, like I mentioned in the opaleye issue.

**Comment by c-lewis** [comment]
Is this issue the reason why the Opaleye dependency has not been changed to allow 0.4?

**Comment by bergmark** [comment]
Yep, Everything else should work with 0.4. My suggestion for opaleye [is here](https://github.com/tomjaguarpaw/haskell-opaleye/issues/92#issuecomment-143681655). I forgot to ping tom again, I'll do that now.
